libbeat/cfgfile/reload.go

```diff
@@ -15,3 +31,163 @@ type ReloadConfig struct {
 	Period  time.Duration `config:"period"`
 	Enabled bool          `config:"enabled"`
 }
+
+type RunnerFactory interface {
+	Create(*common.Config) (Runner, error)
+}
```

Comment:
I wonder who created this common pattern ;-) I personally prefer the interface one.